include/CppInterOp/CppInterOp.h

```diff
@@ -281,6 +281,9 @@ CPPINTEROP_API bool IsNamespace(TCppScope_t scope);
 /// Checks if the scope is a class or not.
 CPPINTEROP_API bool IsClass(TCppScope_t scope);
 
+/// Checks if the scope is a CUDA function or not.
+CPPINTEROP_API bool IsCUDAFunction(TCppScope_t scope);
```

Comment:
Can we generalize this api to be "HasAnnotation(AnnotationKind [ | AnnotationKind...] )"